sonar-php-plugin/src/main/java/org/sonar/plugins/php/phpunit/xml/TestSuite.java

```diff
@@ -104,102 +80,62 @@ public TestSuite(final String name, final String file, final String tests, final
     this.testCases = testCases;
   }
 
-  /**
-   * Gets the name.
-   *
-   * @return the name
-   */
-  public String getName() {
-    return name;
-  }
-
-  /**
-   * Sets the name.
-   *
-   * @param name the new name
-   */
   public void setName(final String name) {
     this.name = name;
   }
 
-  /**
-   * Gets the file.
-   *
-   * @return the file
-   */
   public String getFile() {
     return file;
   }
 
-  /**
-   * Sets the file.
-   *
-   * @param file the new file
-   */
   public void setFile(final String file) {
     this.file = file;
   }
 
-  /**
-   * Gets the tests.
-   *
-   * @return the tests
-   */
-  public String getTests() {
-    return tests;
-  }
-
-  /**
-   * Sets the tests.
-   *
-   * @param tests the new tests
-   */
-  public void setTests(final String tests) {
-    this.tests = tests;
-  }
-
-  /**
-   * Gets the assertions.
-   *
-   * @return the assertions
-   */
   public String getAssertions() {
     return assertions;
   }
 
-  /**
-   * Sets the assertions.
-   *
-   * @param assertions the new assertions
-   */
   public void setAssertions(final String assertions) {
     this.assertions = assertions;
   }
 
-  /**
-   * Gets the time.
-   *
-   * @return the time
-   */
   public double getTime() {
     return time;
   }
 
-  /**
-   * Gets the test suites.
-   *
-   * @return the test suites
-   */
-  public List<TestSuite> getTestSuites() {
-    return testSuites;
+  public List<TestCase> getTestCases() {
+    return testCases;
+  }
+
+  public boolean isFileBasedSuite() {
+    return getFile() != null;
   }
 
   /**
-   * Gets the test cases.
+   * Reads the given test suite.
+   * <p/>
+   * Due to a inconsistent XML format in phpUnit, we have to importReport enclosing testsuite name for generated testcases when a testcase holds
+   * the annotation dataProvider.
    *
-   * @return the test cases
-   */
-  public List<TestCase> getTestCases() {
-    return testCases;
+   * @param activeReport
+   */
+  public Collection<PhpUnitTestFileReport> generateReports(@Nullable PhpUnitTestFileReport activeReport) {
+    Map<String, PhpUnitTestFileReport> reportsPerFile = new HashMap<>();
+    if (isFileBasedSuite()) {
+      reportsPerFile.putIfAbsent(getFile(), new PhpUnitTestFileReport(getFile(), getTime()));
+      activeReport = reportsPerFile.get(getFile());
+    }
+    for (TestCase testCase : getTestCases()) {
+      if (activeReport != null) {
+        activeReport.addTestCase(testCase);
+      } else {
+        LOGGER.warn("Test cases must always be descendants of a file-based suite, skipping : " + testCase.fullName() + " in " + name);
+      }
+    }
+    for (TestSuite childSuite : this.testSuites) {
```

Comment:
They were vestigial methods (from when the generateReports method was in the "suites reader"). Removed bost of them.